tests/cli/main.cpp

```diff
@@ -0,0 +1,365 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <graphene/app/application.hpp>
+#include <graphene/app/plugin.hpp>
+
+#include <graphene/utilities/tempdir.hpp>
+
+#include <graphene/account_history/account_history_plugin.hpp>
+#include <graphene/witness/witness.hpp>
+#include <graphene/market_history/market_history_plugin.hpp>
+#include <graphene/egenesis/egenesis.hpp>
+#include <graphene/wallet/wallet.hpp>
+
+#include <fc/thread/thread.hpp>
+#include <fc/smart_ref_impl.hpp>
+#include <fc/network/http/websocket.hpp>
+#include <fc/rpc/websocket_api.hpp>
+#include <fc/rpc/cli.hpp>
+
+#include <boost/filesystem/path.hpp>
+
+#define BOOST_TEST_MODULE Test Application
+#include <boost/test/included/unit_test.hpp>
+
+/*********************
+ * Helper Methods
+ *********************/
+
+// hack:  import create_example_genesis() even though it's a way, way
+// specific internal detail
+namespace graphene { namespace app { namespace detail {
+	graphene::chain::genesis_state_type create_example_genesis();
+} } } // graphene::app::detail
+
+boost::filesystem::path create_genesis_file(fc::temp_directory& directory) {
+	boost::filesystem::path genesis_path = boost::filesystem::path{directory.path().generic_string()} / "genesis.json";
+    fc::path genesis_out = genesis_path;
+    graphene::chain::genesis_state_type genesis_state = graphene::app::detail::create_example_genesis();
+    std::cerr << "Creating example genesis state in file " << genesis_out.generic_string() << "\n";
+    fc::json::save_to_file(genesis_state, genesis_out);
+    return genesis_path;
+}
+
+/***
+ * @brief Start the application, listening on port 8090
+ * @param app_dir the temporary directory to use
+ * @returns the application object
+ */
+std::shared_ptr<graphene::app::application> start_application(fc::temp_directory& app_dir) {
+	std::shared_ptr<graphene::app::application> app1(new graphene::app::application{});
+
+    app1->register_plugin<graphene::account_history::account_history_plugin>();
+    app1->register_plugin< graphene::market_history::market_history_plugin >();
+    app1->register_plugin< graphene::witness_plugin::witness_plugin >();
+    app1->startup_plugins();
+    boost::program_options::variables_map cfg;
+    cfg.emplace("rpc-endpoint", boost::program_options::variable_value(string("127.0.0.1:8090"), false));
+    cfg.emplace("genesis-json", boost::program_options::variable_value(create_genesis_file(app_dir), false));
+    cfg.emplace("seed-nodes", boost::program_options::variable_value(string("[]"), false));
```

Comment:
No longer attempting to connect to seed nodes during test

Reply:
good, thanks.